client_test/helpers_test.go

```diff
@@ -229,3 +231,22 @@ func closeSimulatedChain(t *testing.T, chain chainservice.SimulatedChain) {
 		t.Fatal(err)
 	}
 }
+
+// generateMessageKey generates a ECDSA private key deterministically using the given pk bytes.
+func generateMessageKey(pk []byte) p2pcrypto.PrivKey {
+
+	// We use he given
+	totalSize := 256
+	large := make([]byte, totalSize)
+	copy(large, pk)
+
+	messageKey, _, err := p2pcrypto.GenerateECDSAKeyPair(bytes.NewReader(large))
+	if err != nil {
+		panic(err)
```

Comment:
Should we return the error?

Reply:
Since this is purely a test helper I think it's OK to `panic`?

Reply:
Alternatively it could accept a `t *testing.T` and call `t.Fatal`

Reply:
I like the second alternative!

Reply:
Updated in https://github.com/statechannels/go-nitro/pull/1194/commits/be5ea3376db8a084983debe20dccf5b9ebeeac69